Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -902,7 +898,7 @@ extension TaskHandler: ChannelDuplexHandler {
 
     private func writeBodyPart(context: ChannelHandlerContext, part: IOData, promise: EventLoopPromise<Void>) {
         switch self.state {
-        case .idle:
+        case .sending:
```

Comment:
this seems a little dangerous? Previously we could send in `.idle` and now we can't?

Reply:
A full `switch` without the `default` would be better too here.

Reply:
done